### PR TITLE
Fix and add unit tests

### PR DESCRIPTION
**Pull request by lewismarshall**
This PR fixes / simplifies core logic and adds some unit tests.

Also an issue with the cmd entry-point issue is resolved for now (by not returning and running as a service after startup).

Currently this PR includes PR #36.

### CODE REVIEW

**Comment by vaijab** [comment]
Looks like a broken build?

**Comment by lewismarshall** [comment]
Drone seems to be broken or running very slow. I'll check when back in...

File: pkg/kmm/kmm.go [hunk 1]
```diff
@@ -102,6 +102,9 @@ func GetAssets(cfg Config) (err error) {
 		if err = kubeadm.CreateKubeConfig(cfg.KubeadmCfg); err != nil {
 			return err
 		}
+		if err = CreateAndStartKubelet(cfg.KubeadmCfg.CloudProvider, cfg.KubeadmCfg.KubeVersion, true); err != nil {
```
Comment (vaijab) [inline pkg/kmm/kmm.go]: Where is `err` initialized?

File: pkg/kmm/kubelet.go [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+	"github.com/UKHomeOffice/keto-k8/pkg/fileutil"
+	"github.com/UKHomeOffice/keto-k8/pkg/constants"
+)
+
+// CreateAndStartKubelet will create Kubelet
+func CreateAndStartKubelet(cloudProvider, kubeVersion string, master bool) error {
+
+	// Render kubelet.service
+	data := struct {
+		CloudProviderName	string
```
Comment (vaijab) [inline pkg/kmm/kubelet.go]: Indentation.

**Comment by lewismarshall** [comment]
@vaijab fixed-up your changes in PR #36 and rebased this of that. Fixed the build - was a timeout, added service flag and updated tests to set appropriately.

File: pkg/kmm/kmm.go [hunk 3]
```diff
@@ -18,207 +18,267 @@ import (
...
+	Kmm                  Interface
 }
 
-// Manifests - Will generate static manefest files
-func Manifests(cfg kubeadm.Config) (err error) {
-	if err = kubeadm.WriteManifests(cfg); err != nil {
+// Both structs here use the same config but are bound to different methods...
+
+// Config is tied to the Primary methods (no interface - not for mocking)
+type Config struct {
```
Comment (vaijab) [inline pkg/kmm/kmm.go]: I don't get this? Why embed `ConfigType` struct here?

Comment (lewismarshall) [inline pkg/kmm/kmm.go]: Maybe a better name required but essentially there are high level methods (entry-points) and then low-level methods (mockable bits) - all of which use the same config. This allows a simple type cast to move the data...

**Pull request merged**